### ISSUE TITLE
Cannot access members of class in with statement

### ISSUE THREAD

**Issue by MarkusVidefors** [microsoft/pyright#1976]
Hi,

Just installed pyright for coc-nvim yesterday and ran into a weird error when editing some old code. If there's something I missed I'll be happy to provide more info.

**Describe the bug**
The bug is that pyright cannot access members of my class when used in a with statement. Outside of a with statement it detects them correctly (jump to definition and all that works). The class i have inherits from requests.Session and it seems to me that pyright does detect all methods related to requests.Session but none from my own class. So my assumption is that pyright incorrectly believes the type is requests.Session instead of my own.

The error message i recieve for the code below is:
[Pyright reportGeneralTypeIssues] [E] Cannot access member "some_method" for type "Session"
  Member "some_method" is unknown"

As i added the above I realised that it even says the type is Session when it in fact should be Test. My real code uses the name Session so I did not realise this before.

**To Reproduce**
Write a small class inheriting from requests.Session and add a new method. Then create an instance of this session with the with statement and try to access the method you added. 

**Expected behavior**
I expect pyright to detect the methods of the class since to my knowledge the code is correct.

**Screenshots or Code**
This code is what I created to reproduce the results. Unfortunately i cannot share the real code.
```python
import requests

class Test(requests.Session):
    def __init__(self, testing):
        super().__init__()
        self.testing = testing

    def some_method(self):
        print(self.testing)


test = Test("TestString")
test.some_method()  # No error message
test.request("GET", "https://a.url.tld/hi")  # No error message

with Test("aString") as session:
    session.some_method()  # Error message on this row
    session.request("GET", "https://a.url.tld/hi")  # This is fine
```

**VS Code extension or command-line**
I'm running coc-pyright for coc-nvim in neovim.
coc-pyright - v1.1.148
coc-nvim - v0.0.80-b427b2ccfa
nvim - v0.4.4

**Additional context**
I haven't tried to recreate this bug without usage of the requests library but if that is needed I could certainly give it a try.

**Comment by JelleZijlstra** [comment]
This is a bug in typeshed: https://github.com/python/typeshed/blob/master/stubs/requests/requests/sessions.pyi#L77 should use a bound typevar. I'll submit a PR.

**Comment by erictraut** [comment]
Thanks @JelleZijlstra for the quick fix. I've pulled the latest typeshed stubs into pyright and confirmed that the error in the above code is no longer reported. The updated stubs will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.149, which I just published. It will also be in the next release of pylance.
